index/generator/library/library.go

```diff
@@ -193,6 +222,25 @@ func validateIndexComponent(indexComponent schema.Schema, componentType schema.D
 	if len(indexComponent.Architectures) == 0 {
 		return &MissingArchError{devfile: indexComponent.Name}
 	}
+	if len(indexComponent.DeploymentScopes) > 2 {
+		return &TooManyDeploymentScopes{devfile: indexComponent.Name}
+	}
+	for kind := range indexComponent.DeploymentScopes {
+		if kind != schema.InnerloopKind && kind != schema.OuterloopKind {
+			return &InvalidDeploymentScopes{devfile: indexComponent.Name, deploymentScopeKind: kind}
+		}
+	}
+	for _, version := range indexComponent.Versions {
+		if len(version.DeploymentScopes) > 2 {
```

Comment:
@michael-valdron I was wondering what would be the case that a version will have 2 deployment scopes. I'm thinking this could be innerloop or outerloop right?

Reply:
I guess is based on the same format like CommandGroups right?

Reply:
> @michael-valdron I was wondering what would be the case that a version will have 2 deployment scopes. I'm thinking this could be innerloop or outerloop right?

@thepetk Correct, every version (and stack) can only have 2 deployment scopes max, innerloop and outerloop.